### ISSUE TITLE
How to invalidate build cache?

### ISSUE THREAD

**Issue by kkesley** [aws-amplify/amplify-console#520]
** Please describe which feature you have a question about? **

Amplify build process

** Provide additional details**

I'm currently seeing failing builds on my Amplify project (React). I believe this is because we rewrote our react app from JS to TS. It complains about unsupported syntax because we upgraded some packages in our project. I believe I've seen this problem when we used CodeBuild and we solved it by using the [`invalidate-project-cache`](https://docs.aws.amazon.com/cli/latest/reference/codebuild/invalidate-project-cache.html) command.

Is there a similar command from Amplify?

**Comment by kkesley** [comment]
We can do `rm -rf node_modules` in the build step for now. But it would be nice if we can invalidate the cache from outside the build

**Comment by kahdojay** [comment]
Hi @kkesley, this isn't something we support but your feedback is helpful, we'll take this as a feature request. Glad you were able to unblock yourself in the meantime, closing this for now, feel free to reopen if you have any other questions!

**Comment by hirenumradia** [comment]
+1 for this, I ran into this issue today. It would be great to disable the build cache and specify specific paths inside amplify.yml.

**Comment by davedotdev** [comment]
+1 for this feature too. I've ran into headaches where I've upgraded from a trial package to a commercial package and instead of the build process grabbing the packages from a private repository, the trial ones have been used, despite the `.npmrc` file being there. I even tried doing manual login steps in the build file, but nada. As soon as I removed the cache parameters, it worked just fine. Since my initial head scratch moment, I raised a ticket with AWS support and they pointed me here. Being able to purge the build cache from outside the pipeline would be great in troubleshooting things like this. I lost a few hours to this problem. You can remove packages from the build env with commands, but it means having to mess around with the pipeline config until you get to a happy build outcome.

**Comment by natBizitza** [comment]
Nice to have feature! Having issues seeing deployed changes Using Nextjs 12 and Amplify.

**Comment by andreadeluisi-ryte** [comment]
Same here, working with a Lerna monorepo + Storybook using Amplify builds. It fails too often - I guess we can say randomly - probably due to a caching problem of cross-package dependencies. The solution mentioned above `rm -rf node_modules` looks like a good workaround but I would have preferred having the possibility to control the build-caching on IaC rather than an additional command on the .yml file. 
I'll be so glad to have an update or a better suggestion.

**Comment by gustavostachera** [comment]
+1 We faced a similar issue when migrating our application from Amplify 4.x to 5.x, and to deal with it, we changed our build settings, removing the node_modules to fix it. However, as mentioned, this is not a good approach, because we need to revert this change every time or disable the cache. It'll really be nice if we could do this through a button in the AWS console.

**Comment by sisada-luzern** [comment]
+1 This feature would be really helpful.
